assets/js/Model.js

```diff
@@ -0,0 +1,41 @@
+global.App = require( './App.js' );
+global.OO = require( '../../node_modules/oojs/dist/oojs.js' );
+
+/**
+ * @constructor
+ * @param {Object} translations All existing translation strings.
+ */
+App.Model = function appModel( translations ) {
+	OO.EventEmitter.call( this );
+	this.translations = translations;
+	this.sourceLang = null;
+};
+
+OO.mixinClass( App.Model, OO.EventEmitter );
+
+App.Model.prototype.setSourceLang = function ( lang ) {
+	this.sourceLang = lang;
```

Comment:
In the specific case of source lang setting it's probably not a huge must, but generally, it's good practice in setters that emit events to wrap the event with a conditional to make sure it is only emitted if the value actually changed. This prevents event loops, where an event triggers a change that emits an event that triggers a change that emits.... etc ;)

So, for example, something like

```
if ( this.sourceLang !== lang ) {
  this.sourceLang = lang;
  this.emit( 'sourceLangSet' );
}
```

Also a generally good idea for any other setter that emits an event, for logic and safety.

Reply:
By the way, I just thought of something -- you can also emit the event *with* the language data, to save ourselves the "ask the model what the event state is"

For instance --

this.emit( 'sourceLangSet', this.sourceLang );

Then in the listener, we can get the data by default -- `function onSourceLangSet( currLang ) { ... }`